api/orpy.py

```diff
@@ -0,0 +1,20 @@
+from collections import namedtuple
+
+Application = namedtuple(
```

Comment:
any reason this is a `namedtuple` instead of other data structure ?

Reply:
Last time I benchmarked it was 20% faster than a class, it is also easier to review than a class. Also, it should be explicit that Application instance is immutable. The latter make it less likely to have non-stateless application service, and side-effects that are not visible in another instance of the service, except for local cache, that is not present at this time.

The use of only `namedtuple` and `class` avoids the need to enter into yet-other-record-object-interfaces that is way too complex (dataclass are too complex), it keeps the number of concepts to be known, and be aware of to a minimum low of two that can be composed.

In this case, a dataclass does not bring any useful feature.

TLDR: The rationale is KISS: a `collections.namedtuple` is simple.